### PR TITLE
Changed File upload logic to handle large file size and improved error handling 

### PR DESCRIPTION
**Pull request by Ushmajit**
### Changes

1. Improved error handling in the main module.
2. Changed file upload logic to handle large file sizes. Reference : https://stackoverflow.com/questions/37336050/pipe-a-stream-to-s3-upload
3. Created a driver file to easily test the module functionality.

### Testing
E-2-E testing using driver.js

### CODE REVIEW

File: src/aws_s3_client_module.js [hunk 1]
```diff
@@ -19,46 +20,29 @@ const {S3, Endpoint} = pkg;
...
+        };
+    };
+    const filePath = path.basename(file);
+    const { writeStream, promise } = uploadStream({Bucket: bucket, Key: filePath});
+    fsExtra.createReadStream(filePath).pipe(writeStream);
+    let response = {
+        status: true,
+        errorMessage: ''
+    };
+    await promise.catch((reason)=> { response.status = false; response.errorMessage(reason.toString()); });
```
Comment (abose) [inline src/aws_s3_client_module.js]: Good to split into multiple lines to step with debugger to debug error messages. {} to 
{
}.

File: src/driver.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/** Driver file for the module, uncomment this file to test the changes
```
Comment (abose) [inline src/driver.js:1]: Could move this to mocha unit/integ tests when we have tests later.

File: src/aws_s3_client_module.js (hunk 1 above)
Comment (abose) [reply src/aws_s3_client_module.js]: Will this line succeed for both success and exception case?

Comment (Ushmajit) [reply src/aws_s3_client_module.js]: yup tested it out using the code on driver.js, intentionally manipulated the params to throw an exception, Got the following message

Error uploading the file: test.zip Message: UnknownEndpoint: Inaccessible host:
`bucket_region.linodeobjects.com' at port `undefined'. This service may not be a
vailable in the `us-east-1' region.

File: src/driver.js (hunk 2 above)
Comment (Ushmajit) [reply src/driver.js:1]: Sure, this is temporary till we have tests in place.

**Comment by Ushmajit** [comment]
@abose Addressed your comments in the next commit

**Pull request merged**